erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -727,20 +770,47 @@ def get_transfered_raw_materials(self):
 
 		po_qty = frappe.db.sql("""select qty, produced_qty, material_transferred_for_manufacturing from
 			`tabWork Order` where name=%s""", self.work_order, as_dict=1)[0]
+
 		manufacturing_qty = flt(po_qty.qty)
 		produced_qty = flt(po_qty.produced_qty)
 		trans_qty = flt(po_qty.material_transferred_for_manufacturing)
 
 		for item in transferred_materials:
 			qty= item.qty
+			req_items = frappe.get_all('Work Order Item',
+				filters={'parent': self.work_order, 'item_code': item.item_code},
+				fields=["required_qty", "consumed_qty"]
+			    )
+			req_qty = flt(req_items[0].required_qty)
+			req_qty_each = flt(req_qty / manufacturing_qty)
+			consumed_qty = flt(req_items[0].consumed_qty)
+
+			if trans_qty and manufacturing_qty >= (produced_qty + flt(self.fg_completed_qty)):
+				if qty >= req_qty:
+					qty = (req_qty/trans_qty) * flt(self.fg_completed_qty)
+				else:
+					qty = qty - consumed_qty
+
+				if self.purpose == 'Manufacture':
+					# If Material Consumption is booked, must pull only remaining components to finish product
```

Comment:
The same calculation is applicable if "Backflush Raw Materials Based On" is set as "BOM" in Manufacturing Settings. It should consider Consumption entry and pull only remaining items.

Reply:
@nabinhait this is what this code means.

Reply:
But I think it does not work if "Backflush Raw Materials Based On" is set as "BOM" in Manufacturing Settings. I think `get_transfered_raw_materials` is not called for that use case.